src/test/java/picard/sam/SamErrorMetric/CollectSamErrorMetricsTest.java

```diff
@@ -357,5 +530,335 @@ public void testTooManyDirectives() {
+                { new String[] { "50M2D50M" },      ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 0, 1) },
+                { new String[] { "20M2D20M2D20M" }, ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 0, 2) },
+                { new String[] { "1D10M" },         ".indel_error_by_indel_length", 30, new IndelErrorMetric("1", 0, 0, 1) },
+                { new String[] { "10M1D" },         ".indel_error_by_indel_length", 30, new IndelErrorMetric("1", 0, 0, 1) },
+                // insertions & deletions
+                { new String[] { "20M2I20M3D20M" }, ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 1, 0) },
+                { new String[] { "20M2I20M3D20M" }, ".indel_error_by_indel_length", 30, new IndelErrorMetric("3", 0, 0, 1) },
+                { new String[] { "2I2D20M" },       ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 1, 1) },
+                { new String[] { "2D2I20M" },       ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 1, 1) },
+                { new String[] { "2M2D2I" },        ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 1, 1) },
+                { new String[] { "20M2I2D" },       ".indel_error_by_indel_length", 30, new IndelErrorMetric("2", 0, 1, 1) },
+
+                // is_insertion:
+                { new String[] { "20M1I20M" },      ".indel_error_by_is_insertion", 30, new IndelErrorMetric("true", 0, 1, 0) },
+                { new String[] { "2I2D20M" },       ".indel_error_by_is_insertion", 30, new IndelErrorMetric("true", 0, 1, 0) },
+                { new String[] { "2D2I20M" },       ".indel_error_by_is_insertion", 30, new IndelErrorMetric("true", 0, 1, 0) },
+                { new String[] { "2M2D2I" },        ".indel_error_by_is_insertion", 30, new IndelErrorMetric("true", 0, 1, 0) },
+                { new String[] { "20M2I2D" },       ".indel_error_by_is_insertion", 30, new IndelErrorMetric("true", 0, 1, 0) },
+                // is_deletion
+                { new String[] { "20M1D20M" },      ".indel_error_by_is_deletion", 30, new IndelErrorMetric("true", 0, 0, 1) },
+                { new String[] { "2I2D20M" },       ".indel_error_by_is_deletion", 30, new IndelErrorMetric("true", 0, 0, 1) },
+                { new String[] { "2D2I20M" },       ".indel_error_by_is_deletion", 30, new IndelErrorMetric("true", 0, 0, 1) },
+                { new String[] { "2M2D2I" },        ".indel_error_by_is_deletion", 30, new IndelErrorMetric("true", 0, 0, 1) },
+                { new String[] { "20M2I2D" },       ".indel_error_by_is_deletion", 30, new IndelErrorMetric("true", 0, 0, 1) },
+
+        };
+    }
+
+    @Test(dataProvider = "provideForTestIndelErrors")
+    public void testIndelErrors(final String[] readCigars, final String errorSubscript, final int priorQ, IndelErrorMetric expectedMetric) throws IOException {
```

Comment:
since all your test-cases use priorQ=30, you could remove that from the signature.